content/3_Silver/More_Prefix_Sums.mdx

```diff
@@ -652,6 +652,74 @@ for _ in range(query_num):
 
 <Problems problems="cum2" hideSuggestProblemButton />
 
+## Difference Arrays
+
+<Resources>
+	<Resource source="Codeforces" title="An Introduction To Difference Arrays" starred />
+</Resources>
+
+<FocusProblem problem="sample3" />
+
+### Explanation
+
+We need to know how many times each operation is applied i.e. how many times is contained within a query interval.
+For this, maintain a frequency array $s$; $s[i]$ is the number of times operation $i$ is applied. The important
+step is how the frequency array is updated. Instead of iterationg through the interval $[l, r]$ and incrementing
+each value by one, which would result in $\mathcal{O}(K \cdot M)$ time complexity, we could just increment
+$s[l]$ by one, the left bound, and decrement by one $s[r+1]$, the cell next to the right bound. After processing the queries
+we obtain the frequency array by doing prefix sums on $s$, resulting in $\mathcal{O}(M)$ time complexity. The second part,
+applying the operations, can be done exactly the same way. This technique is called difference array and has linear time complexity
+coming from interating through the array for making prefix sums.
+
+### Implementation
+
+**Time Complexity:** $\mathcal{O}(N+M)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <tuple>
+#include <vector>
+
+using namespace std;
+
+int main() {
+	int n, m, k;
+	cin >> n >> m >> k;
+	vector<int> a(n + 1);
+	for (int i = 1; i <= n; i++) { cin >> a[i]; }
+	vector<tuple<int, int, int>> updates(m);
+	for (tuple<int, int, int> &update : updates) {
```

Comment:
hmm, kind of agree with envy. is there any reason you chose to use tuples instead of vector/std::array

Reply:
Idk, It seems to me a little be weird to use vector/array for just 3 values, but if you consider it strange i'll change it